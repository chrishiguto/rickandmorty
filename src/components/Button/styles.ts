import styled, { css } from 'styled-components'

export const Wrapper = styled.button`
  ${({ theme }) => css`
    background: transparent;
    padding: ${theme.spacings.xxsmall} calc(${theme.spacings.xsmall} - 0.2rem);
    font-size: ${theme.font.sizes.medium};
    font-weight: ${theme.font.bold};
    border: 2px solid ${theme.colors.white};
    border-radius: ${theme.border.radius};
    color: ${theme.colors.white};
    outline: none;
    transition: box-shadow, background-color ${theme.transition.default};

    &:hover {
      background-color: ${theme.colors.darkGray}35;
    }

    &:focus-visible {
      box-shadow: 0 0 0 0.3rem ${theme.colors.yellow};
    }
  `}
`
