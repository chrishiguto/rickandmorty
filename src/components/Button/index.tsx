import { ButtonHTMLAttributes } from 'react'

import * as S from './styles'

type ButtonType = ButtonHTMLAttributes<HTMLButtonElement>

export type ButtonProps = {
  children: React.ReactNode
} & ButtonType

const Button = ({ children, ...props }: ButtonProps) => (
  <S.Wrapper {...props}>{children}</S.Wrapper>
)

export default Button
