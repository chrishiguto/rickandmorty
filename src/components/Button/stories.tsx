import { Story, Meta } from '@storybook/react/types-6-0'
import Button, { ButtonProps } from '.'

export default {
  title: 'Button',
  component: Button,
  parameters: {
    backgrounds: {
      default: 'dark'
    }
  }
} as Meta

export const Default: Story<ButtonProps> = (args) => (
  <Button {...args}>Close</Button>
)
