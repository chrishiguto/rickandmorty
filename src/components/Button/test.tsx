import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import Button from '.'

describe('<Button />', () => {
  it('should render the button', () => {
    const { container } = renderWithTheme(<Button>Button text</Button>)

    expect(
      screen.getByRole('button', { name: /button text/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
