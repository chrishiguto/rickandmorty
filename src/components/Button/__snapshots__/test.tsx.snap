// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render the button 1`] = `
.c0 {
  background: transparent;
  padding: 0.8rem calc(1.6rem - 0.2rem);
  font-size: 1.6rem;
  font-weight: 700;
  border: 2px solid #ffffff;
  border-radius: 0.8rem;
  color: #ffffff;
  outline: none;
  -webkit-transition: box-shadow,background-color 0.3s cubic-bezier(0.4,0,0.2,1);
  transition: box-shadow,background-color 0.3s cubic-bezier(0.4,0,0.2,1);
}

.c0:hover {
  background-color: #8C8C8C35;
}

.c0:focus-visible {
  box-shadow: 0 0 0 0.3rem #cbd736;
}

<button
  class="c0"
>
  Button text
</button>
`
